### PR TITLE
Fixed floating point conversion bug.

### PR DESCRIPTION
**Pull request by VovaViliLox**
For additional information, please refer to the following Asana ticket - https://app.asana.com/0/1202282689536104/1204316175236567/f

### CODE REVIEW

File: lox_services/persistence/database/utils.py [hunk 1]
```diff
@@ -141,3 +143,53 @@ def make_temporary_table(
     table_ref = client.get_table(table_ref)
     table_ref.expires = datetime.now(timezone.utc) + timedelta(hours=1)
     client.update_table(table_ref, ["expires"])  # API request
+
+
+def make_validate_country_code() -> Callable[[pd.DataFrame, str], None]:
+    """Namespace for all the valid country codes."""
+    valid_codes = {country.alpha_2 for country in pycountry.countries}
+
+    def inner_validate_country_code(
```
Comment (melvil-donnart) [inline lox_services/persistence/database/utils.py:152]: @VovaViliLox  Can you add in the docstring the fact that NaN values don't return an error. It looks like it does with your current kne

File: lox_services/persistence/database/push_invoices_data.py [hunk 2]
```diff
@@ -58,16 +89,25 @@ def push_run_to_database(
...
-                if column not in df_invoice.columns:
-                    df_invoice.loc[:, column] = None
 
-            # "Make sure that the columns have the good type
-            df_invoice = df_invoice.fillna(value=na_invoices).astype(dtypes_invoices)
+        postal_code_cols = df_invoice.columns[
+            df_invoice.columns.isin({"postal_code_receiver", "postal_code_sender"})
+        ].tolist()
+        # In case postal_code_reciever falsely gets interpreted as a float
+        df_invoice[postal_code_cols] = (
```
Comment (melvil-donnart) [inline lox_services/persistence/database/push_invoices_data.py:99]: @VovaViliLox Can we have this in a small function?

Comment (melvil-donnart) [reply lox_services/persistence/database/push_invoices_data.py:99]: Both the remove of .0 and the validation of the country code cna be named

process_countries_column()

File: lox_services/persistence/database/utils.py [hunk 3]
```diff
@@ -141,3 +143,53 @@ def make_temporary_table(
...
+            .apply(lambda col: (col.notna()) & ~(col.isin(valid_codes)))
+            .any(axis="columns")
+        )
+        if error_mask.any():
+            print(
+                df.loc[error_mask, country_code_col].pipe(
+                    tabulate, headers="keys", tablefmt="psql"
+                )
+            )
+            raise ValueError(
```
Comment (melvil-donnart) [inline lox_services/persistence/database/utils.py:187]: Can we hanlde NaN values not to raise a ValueError as this is not always specified by DHL, UPS ..

**Review by melvil-donnart — CHANGES_REQUESTED** [review]
Great work! Close to completion

File: lox_services/persistence/database/push_invoices_data.py (hunk 2 above)
Comment (VovaViliLox) [reply lox_services/persistence/database/push_invoices_data.py:99]: If you prefer it this way - sure.

File: lox_services/persistence/database/utils.py (hunk 3 above)
Comment (VovaViliLox) [reply lox_services/persistence/database/utils.py:187]: Empty values get evaluated as False due to AND condition.

File: lox_services/persistence/database/push_invoices_data.py [hunk 4]
```diff
@@ -58,16 +106,11 @@ def push_run_to_database(
...
             infer_datetime_format=date_format,
             header=0,
-            dtype={"postal_code_reciever": str},
-        )
-        if df_invoice.shape[0] > 0:
-            # If some columns are missing, add the columns with None values
-            for column in dtypes_invoices:
-                if column not in df_invoice.columns:
-                    df_invoice.loc[:, column] = None
+            dtype={"postal_code_reciever": "string[pyarrow]"},
```
Comment (melvil-donnart) [inline lox_services/persistence/database/push_invoices_data.py:109]: postal_code_reciever doesn't exist
why postal_code_sender not also included?

**Review by melvil-donnart — CHANGES_REQUESTED** [review]

File: lox_services/persistence/database/push_invoices_data.py (hunk 4 above)
Comment (VovaViliLox) [reply lox_services/persistence/database/push_invoices_data.py]: Typo fixed.

File: lox_services/persistence/database/utils.py [hunk 5]
```diff
@@ -2,9 +2,11 @@
 import os
 import re
 from datetime import datetime, timedelta, timezone
-from typing import Literal
+from typing import Callable, Literal, Sequence, Union
 
+from tabulate import tabulate
 import pandas as pd
+import pycountry
```
Comment (melvil-donnart) [inline lox_services/persistence/database/utils.py:9]: Add to requirements.txt

Comment (VovaViliLox) [reply lox_services/persistence/database/utils.py:9]: requirements.txt, line 36 - `pycountry >= 22.3.5`

**Review by melvil-donnart — APPROVED** [review]
@natasazekic-loxsolution I let your merge on lox_services?

**Pull request merged**